build/helper/metadata_filters.py

```diff
@@ -270,13 +270,13 @@ def filter_ivi_dance_parameter(function):
         Parameter dict if one is found
     '''
     params = filter_parameters(function, ParameterUsageOptions.IVI_DANCE_PARAMETER)
-    if len(params) == 0:
-        return None
-    assert len(params) == 1, 'Found several ivi-dance parameters, expected only one. Found: {0}'.format([x['name'] for x in params])
-    return params[0]
+    if len(params) > 0:
+        size_param = params[0]['size']['value']
+        assert all(x['size']['value'] == size_param for x in params)
```

Comment:
It's worth explaining this, either here or on the function documentation. It can be done in the assert itself.

Something like "if we have multiple ivi-dance parameters, they all must be of the same size."